From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
Date: Thu, 17 Aug 2017 20:13:41 +0300
Subject: [ARCHEOLOGY] Enable DT overlay support in rk3328 u-boot

> X-Git-Archeology: - Revision 64f5514ea06c30ecd9be8b3a994c52fddbc0c09e: https://github.com/armbian/build/commit/64f5514ea06c30ecd9be8b3a994c52fddbc0c09e
> X-Git-Archeology:   Date: Thu, 17 Aug 2017 20:13:41 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Enable DT overlay support in rk3328 u-boot
> X-Git-Archeology:
> X-Git-Archeology: - Revision 88b43456663d83114a56a0fc21bbb657b85fd0a2: https://github.com/armbian/build/commit/88b43456663d83114a56a0fc21bbb657b85fd0a2
> X-Git-Archeology:   Date: Sun, 20 Aug 2017 16:13:47 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Update rk3328 patches and config
> X-Git-Archeology:
> X-Git-Archeology: - Revision 41b589372f2c1a8450033db78f3d8ed9db1b2196: https://github.com/armbian/build/commit/41b589372f2c1a8450033db78f3d8ed9db1b2196
> X-Git-Archeology:   Date: Sun, 26 Aug 2018 14:14:14 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igor.pecovnik@gmail.com>
> X-Git-Archeology:   Subject: Creating new family rockchip64. 3288 and 3399 was merged in, except Nanopi which stays in RK3399 family with sources attached to FA repository until we manage to join that in. Enabled Nanopc T4, Nanopi M4, Renegade. Tested on Rock64 and Nanopc T4.
> X-Git-Archeology:
> X-Git-Archeology: - Revision c989a9b205ad4d27900b61b941f3f863a5d112e6: https://github.com/armbian/build/commit/c989a9b205ad4d27900b61b941f3f863a5d112e6
> X-Git-Archeology:   Date: Tue, 27 Nov 2018 12:51:39 +0100
> X-Git-Archeology:   From: Przemyslaw Sztoch <psztoch@finn.pl>
> X-Git-Archeology:   Subject: Repaired broken Rock patch for current Kconfig (patch is not synchronized with current codebase,
> X-Git-Archeology:
> X-Git-Archeology: - Revision b202f8824858c945f56a03999b5e058784789222: https://github.com/armbian/build/commit/b202f8824858c945f56a03999b5e058784789222
> X-Git-Archeology:   Date: Tue, 05 Mar 2019 18:20:58 -0500
> X-Git-Archeology:   From: Martin Ayotte <martinayotte@yahoo.ca>
> X-Git-Archeology:   Subject: add missing overlay in RK3399 U-boot
> X-Git-Archeology:
> X-Git-Archeology: - Revision 5a2b2c360b9c26ca4bd0d309af7cd3994fd08b7d: https://github.com/armbian/build/commit/5a2b2c360b9c26ca4bd0d309af7cd3994fd08b7d
> X-Git-Archeology:   Date: Sun, 03 May 2020 19:15:46 +0200
> X-Git-Archeology:   From: Piotr Szczepanik <piter75@gmail.com>
> X-Git-Archeology:   Subject: Prepare rockchip64 for switch to mainline u-boot and switched ROCK Pi 4 (#1934)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 86abac1fd4e169712a44d245cf7adef4ee17c420: https://github.com/armbian/build/commit/86abac1fd4e169712a44d245cf7adef4ee17c420
> X-Git-Archeology:   Date: Tue, 26 Oct 2021 22:14:41 +0200
> X-Git-Archeology:   From: Piotr Szczepanik <piter75@gmail.com>
> X-Git-Archeology:   Subject: Cleanup u-boot configurations for rockchip64 derivatives (#3150)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 2813365dd25e3ad110936cbf014b95b38d7090ec: https://github.com/armbian/build/commit/2813365dd25e3ad110936cbf014b95b38d7090ec
> X-Git-Archeology:   Date: Mon, 07 Nov 2022 21:29:00 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Move known non working rockhip64 boards to previous boot loader (#4392)
> X-Git-Archeology:
---
 arch/arm/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 4567c183fb..eb479660f5 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1926,6 +1926,8 @@ config ARCH_ROCKCHIP
 	imply SYS_NS16550
 	imply TPL_SYSRESET
 	imply USB_FUNCTION_FASTBOOT
+	select OF_LIBFDT
+	select OF_LIBFDT_OVERLAY
 
 config ARCH_OCTEONTX
 	bool "Support OcteonTX SoCs"
-- 
Armbian

